lib/cute_pets.rb

```diff
@@ -6,16 +6,43 @@
 module CutePets
   extend self
 
-  def post_pet()
-    pet = nil
+  def fetch_pet()
     if ENV.fetch('pet_datasource').downcase == 'petfinder'
       pet = PetFetcher.get_petfinder_pet
     else
       pet = PetFetcher.get_petharbor_pet
     end
+  end
+
+  def post_pet()
+    pet = nil
+    pet = fetch_pet()
+    # Check that we haven't tweeted this pet before.
+    history_file = ENV.fetch('history_file')
+    history_filepath = File.expand_path(history_file)
+    if File.file?(history_filepath)
+      num_tries = ENV.fetch('number_tries').to_i
+      history = File.readlines(history_filepath)
+      num_tries.times do |i|
+        new_pet = true
+        history.each do |url|
+          if url.strip == pet[:link]
+            new_pet = false
+            break
+          end
+        end
+	break if new_pet
+        pet = fetch_pet()
+      end
+    end
+
     if pet
+      File.open(history_filepath, 'a') do |f|
```

Comment:
Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.